### PR TITLE
Update some dependencies

### PR DESCRIPTION
**Pull request by marco-c**
We were accumulating too many out of date dependencies. The more we accumulate the more it gets difficult to update them.
This is updating almost all of them to the latest version.

I've had to add the eslint override in the service worker script because of https://github.com/airbnb/javascript/issues/1632.

### CODE REVIEW

**Comment by marco-c** [comment]
Ah, I forgot selenium-webdriver doesn't support Node 4...

File: package.json [hunk 1]
```diff
@@ -29,26 +29,26 @@
...
+    "geckodriver": "^1.10.0",
     "istanbul": "^0.4.2",
     "mkdirp": "^0.5.1",
-    "mocha": "^3.0.2",
+    "mocha": "^4.0.1",
     "portfinder": "^1.0.2",
-    "selenium-assistant": "1.0.0",
-    "selenium-webdriver": "3.0.0-beta-2",
+    "selenium-assistant": "^5.2.0",
+    "selenium-webdriver": "^3.6.0",
```
Comment (gauntface) [inline package.json]: I'm not 100% sure if this is needed. Selenium-assistant is pretty much built to work with a specific selenium-webdriver instance it defines.

https://github.com/GoogleChromeLabs/selenium-assistant/blob/master/package.json#L57

However, that will not help with the Node 4 travis issue,

**Comment by JamesMilnerUK** [comment]
Is it still necessary to support 4.x ? It retires in a few months any how; https://github.com/nodejs/Release

**Comment by marco-c** [comment]
@gauntface I've disabled the tests that use Selenium on Node 4.

**Pull request merged**